traced/tests/src/stopgap/guard_fail.rs

```diff
@@ -230,7 +230,7 @@ fn bench_recursion(b: &mut Bencher) {
 
     // Test that the SIR interpreter can deal with new recursions after a guard failure.
     fn rec(i: u8) -> u8 {
-        if i < 5 {
+        if i < 100 {
```

Comment:
I think this does make little (if not no) difference to the runtime of the stopgap interpreter, since it only ever executes one bytecode before the normal interpreter takes over again. That's probably why you see so little measurable difference. I think one of your previous commits where you had something like
```
for i in 0..1000 {
    si.interpret()
}
```
might be better to measure the stopgap interpreter so it isn't downed out by `ct.execute` calls. But then again, this will only ever test runtime of a single bytecode, and thus isn't even a good representation of the stopgap interpreter in general. I don't immediately have a better idea of testing the stopgap interpreter but I'll think about it in the back of my head.

Reply:
Since `rec` is a recursive call, it effectively is the `for` loop (in disguise) *but* continually calling itself?

Reply:
I don't think this makes a difference to the runtime of the stopgap interpreter though, since each guard failure kicks us out of the interp_step and back to [here](https://github.com/softdevteam/yk/pull/323/files/847091e5b424c57abf64d88f83991fa14598399b..953cee6f3f2abc620b5e7338cb553d0c5de6dd19#diff-e40b2230bb45a5d0ec3f9ea9a7c503d99052f4da3b2cc179d3fe56415f7e79f8R269).

That being said, I think I talked nonsense above. The stopgap interpreter runs only one bytecode of the user interpreter, but that could involve running multiple MIR bytecodes, so `sg.interpret` tests more than I claimed above.

Reply:
> The stopgap interpreter runs only one bytecode of the user interpreter, but that could involve running multiple MIR bytecodes, so sg.interpret tests more than I claimed above.

Yes, my test is designed to test that (although whether it does so successfully is perhaps another matter!).